MySortedSet/homework4/model/Coffee.java

```diff
@@ -0,0 +1,77 @@
+package homework4.model;
+
+import java.util.Objects;
+
+/**
+ * An OOP representation of coffee.
+ * The class implement the Comparable interface, to make it eligible for adding to a MySortedSet
+ * The class describes different preferences of coffee
+ */
+
+public class Coffee implements Comparable<Coffee>{
+    private String coffeeType;
+    private int sugarCubes;
+    private boolean isHot;
+
+    /**
+     * Constructor of Coffee class
+     * Receives objects and initializes all Coffee fields, based on the
+     * information present in this object
+     */
+    public Coffee(String coffeeType, int sugarCubes, boolean isHot) {
+        this.coffeeType = coffeeType;
+        this.sugarCubes = sugarCubes;
+        this.isHot = isHot;
+    }
+
+    public String getCoffeeType() {
+        return coffeeType;
+    }
+
+    public int getSugarCubes() {
+        return sugarCubes;
+    }
+
+    public boolean isHot() {
+        return isHot;
+    }
+
+    @Override
+    public String toString() {
+        return this.coffeeType + " coffee with " + this.sugarCubes  + " sugar cubes ";
+    }
+
+    /**
+     * 2 coffees are considered equal if all their fields are the same, besides boolean isHot.
+     * It is allowed to vary.
+     * @param o the object, to which 'this' is compared
+     * @return return true if the implicit and explicit parameters are equal, false otherwise
+     */
+    @Override
+    public boolean equals(Object o) {
```

Comment:
"Always override hashCode when you override equals." Josh Bloch, Effective Java (3rd edition), item 11
https://github.com/david-sauvage/effective-java-summary